src/Components/Reports/SystemsPDFReport.js

```diff
@@ -1,51 +1,72 @@
-import { PDF_REPORT_PER_PAGE, SYSTEMS_PDF_REPORT_TITLE } from '../../constants';
+import { PDF_REPORT_PER_PAGE, REPORT_NOTIFICATIONS, SYSTEMS_PDF_REPORT_TITLE } from '../../constants';
 import { fetchSystems } from '../../Utilities/api';
 import { formatData, generateFilterText, getSystemsReportFileName } from './Util';
 import propTypes from 'prop-types';
 import { SystemsTablePage } from './Common/SystemsTablePage';
 import { SystemsFirstPage } from './Common/SystemsFirstPage';
+import { useNotification } from './Common/useNotification';
 
-const generateSystemsPDFReport = async (filters, orderBy, orderHow) => {
+export const DownloadSystemsPDFReport = ({ filters, orderBy, orderHow, ...props }) => {
+    const reportFileName = getSystemsReportFileName();
+    const [addNotification, clearNotifications] = useNotification();
+    const { start, success, failure } = REPORT_NOTIFICATIONS;
 
-    const { systemsReportFirstPage, systemsReportRestPages } = PDF_REPORT_PER_PAGE;
+    useEffect(() => {
+        addNotification(start);
+    }, []); // eslint-disable-line react-hooks/exhaustive-deps
 
-    // Table rows
-    const fetchSystemParams = {
-        filters,
-        stateFilter: filters.stateFilter,
-        orderBy,
-        orderHow
-    };
-    const systemsResponse = await fetchSystems(fetchSystemParams);
-    const pdfData = formatData(systemsResponse.data, 'pdf');
+    const generateSystemsPDFReport = async (filters, orderBy, orderHow) => {
 
-    // first page description and data
-    const totalSystems = systemsResponse?.meta?.count;
-    const filterText = generateFilterText(filters);
-    const firstPageData =  pdfData.splice(0, systemsReportFirstPage);
+        const { systemsReportFirstPage, systemsReportRestPages } = PDF_REPORT_PER_PAGE;
 
-    const firstPage = <SystemsFirstPage
-        data={firstPageData}
-        totalSystems={totalSystems}
-        filterText={filterText} />;
+        // Table rows
+        const fetchSystemParams = {
+            filters,
+            stateFilter: filters.stateFilter,
+            orderBy,
+            orderHow
+        };
 
-    const otherPages = [];
+        let systemsResponse;
+        try {
+            systemsResponse = await fetchSystems(fetchSystemParams);
+        }
+        catch {
+            clearNotifications();
+            addNotification(failure);
 
-    while (pdfData.length > 0) {
-        otherPages.push(pdfData.splice(0, systemsReportRestPages));
-    }
+            return [];
+        }
 
-    return [
-        firstPage,
-        ...otherPages.map((systemsPage, index) => <SystemsTablePage key={index} data={systemsPage}  page={index + 1}/>)
-    ];
+        const pdfData = formatData(systemsResponse.data, 'pdf');
 
-};
+        // first page description and data
+        const totalSystems = systemsResponse?.meta?.count;
+        const filterText = generateFilterText(filters);
+        const firstPageData =  pdfData.splice(0, systemsReportFirstPage);
```

Comment:
```suggestion
        const firstPageProps = {
           data: pdfData.splice(0, firstPageCount),
           totalSystems: systemsResponse?.meta?.count,
           filterText: generateFilterText(filters)
        };   
      
```
As these variables are used just as props for `systemsFirstPage`, instead of declaring individual variables, I would prefer to declare a single variable.

Reply:
Thanks for the suggestion @kgaikwad  👍🏻 
I agree and will push the changes in the latest commit.